gitlab/client.py

```diff
@@ -32,6 +32,13 @@
 
 
 class Gitlab:
+
+    # The session object is defined here as long as the RequestsBackend is not
+    # fully implemented. The long-term goal is to wrap all the http
+    # requests/responses by the RequestsBackend class
+
+    session: requests.Session
+
```

Comment:
For other variables in `__init__` we add type hints along with the variable assignment directly, I think we could be consistent here WDYT? The ambiguity comes from `http_backend`, so I would add the default hint there.

```suggestion
```